src/cdk-experimental/column-resize/resize-strategy.ts

```diff
@@ -0,0 +1,185 @@
+  }
+
+  applyMinColumnSize(cssFriendlyColumnName: string, _: HTMLElement, sizeInPx: number): void {
+    const cssSize = coerceCssPixelValue(sizeInPx);
+
+    this._applyProperty(cssFriendlyColumnName, 'min-width', cssSize,
+        sizeInPx !== this.defaultMinSize);
+  }
+
+  applyMaxColumnSize(cssFriendlyColumnName: string, _: HTMLElement, sizeInPx: number): void {
+    const cssSize = coerceCssPixelValue(sizeInPx);
+
+    this._applyProperty(cssFriendlyColumnName, 'max-width', cssSize,
+        sizeInPx !== this.defaultMaxSize);
+  }
+
+  protected getColumnCssClass(cssFriendlyColumnName: string): string {
+    return `cdk-column-${cssFriendlyColumnName}`;
+  }
+
+  ngOnDestroy() {
+    // TODO: Use remove() once we're off IE11.
+    if (this._styleElement && this._styleElement.parentNode) {
+      this._styleElement.parentNode.removeChild(this._styleElement);
+      this._styleElement = undefined;
+    }
+  }
+
+  private _applyProperty(
+      cssFriendlyColumnName: string,
+      key: string,
+      value: string,
+      enable = true): void {
+    const properties = this._getColumnPropertiesMap(cssFriendlyColumnName);
+
+    if (enable) {
+      properties.set(key, value);
+    } else {
+      properties.delete(key);
+    }
+    this._applySizeCss(cssFriendlyColumnName);
+  }
+
+  private _getStyleSheet(): CSSStyleSheet {
+    if (!this._styleElement) {
+      this._styleElement = this._document.createElement('style');
+      this._styleElement.appendChild(this._document.createTextNode(''));
+      this._document.head.appendChild(this._styleElement);
+    }
+
+    return this._styleElement.sheet as CSSStyleSheet;
+  }
+
+  private _getColumnPropertiesMap(cssFriendlyColumnName: string): Map<string, string> {
+    let properties = this._columnProperties.get(cssFriendlyColumnName);
+    if (properties === undefined) {
+      properties = new Map<string, string>();
+      this._columnProperties.set(cssFriendlyColumnName, properties);
+    }
+    return properties;
+  }
+
+  private _applySizeCss(cssFriendlyColumnName: string) {
+    const properties = this._getColumnPropertiesMap(cssFriendlyColumnName);
+    const propertyKeys = Array.from(properties.keys());
+
+    let index = this._columnIndexes.get(cssFriendlyColumnName);
+    if (index === undefined) {
+      if (!propertyKeys.length) {
+        // Nothing to set or unset.
+        return;
+      }
+
+      index = this._indexSequence++;
+      this._columnIndexes.set(cssFriendlyColumnName, index);
+    } else {
+      this._getStyleSheet().deleteRule(index);
+    }
+
+    const columnClassName = this.getColumnCssClass(cssFriendlyColumnName);
+    const tableClassName = this._columnResize.getUniqueCssClass();
+
+    const selector = `.${tableClassName} .${columnClassName}`;
+    const body = propertyKeys.map(key => `${key}:${properties.get(key)}`).join(';');
+
+    this._getStyleSheet().insertRule(`${selector} {${body}}`, index!);
```

Comment:
Wouldn't it be simpler to assign the styles directly on the element, rather than having to keep track of a separate stylesheet?

Reply:
It would be simpler, but I'm doing this as an optimization to avoid doing a dom manipulation for every row in the table.

Reply:
There'd also be the issue of dealing with rows added later on, which this handles nicely.